### PR TITLE
Task_1

### PR DESCRIPTION
**Pull request by danielbilikId**

### CODE REVIEW

**Comment by Nimtso** [comment]
Rename commits to have meaningful names

File: app.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const request = require('supertest');
```
Comment (Nimtso) [inline app.test.js]: Use `imports` not `require`

File: app.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+const app = require("./app");
+
+describe('Test Healthy connection', () => {
```
Comment (Nimtso) [inline app.test.js]: Look at naming tests clauses conventions.

File: app.test.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    it('Test health:', () => {
+        request(app).get('/health').expect(200)
```
Comment (Nimtso) [inline app.test.js]: Separate the request and the assertion

File: app.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import helmet from 'helmet';
+
+const app = express();
+
+app.use(helmet());
+
+app.get('/health', (req:Request, res:Response) => {
+  res.sendStatus(200);
+});
+export default app;
```
Comment (Nimtso) [inline app.ts]: Space between the export and the function.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "hafifa2",
```
Comment (Nimtso) [inline package.json]: Meaningful name

File: package.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    "express": "^4.18.2",
+    "geojson": "^0.5.0",
+    "helmet": "^6.0.1",
+    "mongo": "^0.1.0",
+    "mongoose": "^6.9.1",
+    "mongoose-geojson-schema": "^2.2.4",
+    "node-ts": "^5.1.2",
+    "nodemon": "^2.0.20",
+    "ol": "^7.2.2",
+    "supertest": "^6.3.3"
```
Comment (Nimtso) [inline package.json]: Dev dependency

Comment (Nimtso) [reply package.json]: relevant to other dependencies too

File: package.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "hafifa2",
+  "version": "1.0.0",
+  "description": "",
+  "main": "app.test.js",
```
Comment (Nimtso) [inline package.json]: This isn't your main

File: package.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  "scripts": {
```
Comment (Nimtso) [inline package.json:7]: Add a start script

File: app.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import express, { Express, Request, Response } from 'express';
+import helmet from 'helmet';
+
+const app = express();
+
+app.use(helmet());
+
+app.get('/health', (req:Request, res:Response) => {
+  res.sendStatus(200);
```
Comment (Nimtso) [inline app.ts]: Send, "I am alive"

File: app.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+});
+export default app;
+module.exports = app;
```
Comment (Nimtso) [inline app.ts]: export default app

File: server.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+const http = require('http'); 
+import app from './app';
```
Comment (Nimtso) [inline server.ts]: Space between external

File: server.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+const http = require('http');
```
Comment (Nimtso) [inline server.ts]: Why ?
You are using express, why are you importing http ?

**Review by Nimtso — COMMENTED** [review]
Rename the repository

File: app.test.js (hunk 2 above)
Comment (gthecht) [reply app.test.js]: Still need to change test descriptions

File: app.ts [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import express, { Express, Request, Response } from 'express';
+import helmet from 'helmet';
+
+const app = express();
+
+app.use(helmet());
+
+app.get('/health', (req:Request, res:Response) => {
```
Comment (gthecht) [inline app.ts]: Use eslint

File: babel.config.js [hunk 14]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (gthecht) [inline babel.config.js]: Where are you using babel? We don't usually use babel

File: package.json [hunk 15]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "CheckSoldiersHealth",
+  "version": "1.0.0",
+  "description": "",
+  "main": "app.test.js",
```
Comment (gthecht) [inline package.json]: you main should be `dist/index.js`

File: package.json [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+  "scripts": {
+    "test": "jest --collectCoverage"
+},
+"author": "",
+"license": "ISC",
+"dependencies": {
+    "@types/geojson": "^7946.0.10",
```
Comment (gthecht) [inline package.json]: Remove all the dependencies you aren't using. When you start using them add them, not before

File: package.json [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+    "mongo": "^0.1.0",
+    "mongoose": "^6.9.1",
+    "mongoose-geojson-schema": "^2.2.4",
+    "node-ts": "^5.1.2",
+    "nodemon": "^2.0.20",
+    "ol": "^7.2.2",
+    "supertest": "^6.3.3"
+},
+"devDependencies": {
+    "@babel/core": "^7.21.0",
```
Comment (gthecht) [inline package.json]: Your aren't using babel

File: server.ts (hunk 11 above)
Comment (gthecht) [reply server.ts]: You can use eslint

File: server.ts [hunk 18]
```diff
@@ -0,0 +1,9 @@
+import * as http from 'http'
+import app from './app';
+
+const server = http.createServer(app);
```
Comment (gthecht) [inline server.ts]: don't use http, use just express

File: server.ts [hunk 19]
```diff
@@ -0,0 +1,9 @@
...
+
+const port = process.env.PORT || 3000;
```
Comment (gthecht) [inline server.ts]:
```JS
const port = process.env.PORT ?? 3000;
```

File: server.ts (hunk 18 above)
Comment (gthecht) [reply server.ts]: look [here](https://iamtimsmith.com/blog/creating-a-basic-server-with-express-js/) for how to run express servers.

**Review by gthecht — COMMENTED** [review]
You need to change your commits

File: server.ts (hunk 18 above)
Comment (danielbilikId) [reply server.ts]: this just worked better for the app-server separation. I tried with express as well. cant remember why It didnt work though

File: package.json (hunk 17 above)
Comment (danielbilikId) [reply package.json]: I am. for the jest to typescript configuration. it passes through babel

File: server.ts (hunk 18 above)
Comment (gthecht) [reply server.ts]: Change to express, http is just another dependency, and you don't actually need it

File: README.md [hunk 20]
```diff
@@ -1 +0,0 @@
-# Hafifa-3.0-help-me-pls-help-i-hate-git
```
Comment (gthecht) [inline README.md:1]: Why did you delete the README?

File: configs/.eslintrc.cjs [hunk 21]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended'],
+    parser: '@typescript-eslint/parser',
+    plugins: ['@typescript-eslint'],
+    root: true,
+  };
+
```
Comment (gthecht) [inline configs/.eslintrc.cjs]: Remove the extra empty line

File: configs/babel.config.js [hunk 22]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    presets: [
+      ['@babel/preset-env', {targets: {node: 'current'}}],
+      '@babel/preset-typescript',
+    ],
+  };
+
```
Comment (gthecht) [inline configs/babel.config.js]: Same

File: server/server.ts [hunk 23]
```diff
@@ -0,0 +1,9 @@
+import * as http from 'http'
+import app from './source/app';
+
+const server = http.createServer(app);
+
+const port = process.env.PORT || 3000;
+server.listen(port, () => {
+  console.log(`Server listening on port ${port}`);
+});
```
Comment (gthecht) [inline server/server.ts]: empty line

File: source/app.ts [hunk 24]
```diff
@@ -0,0 +1,12 @@
...
+
+const app = express();
+
+app.use(helmet());
+
+app.get('/health', (req:Request, res:Response) => {
+  res.send("I am alive!");
+});
+
+export default app;
```
Comment (gthecht) [inline source/app.ts]: empty line

File: tests/app.test.js [hunk 25]
```diff
@@ -0,0 +1,8 @@
+import request from "supertest";
+import  app from "./app";
+
+describe('Is Soldier Alive', () => {
+    it('Should Send an API Get Request If Soldier Is Alive, Expected 200:', () => {
+        request(app).get('/health').expect("I am alive!");
+    });
+});
```
Comment (gthecht) [inline tests/app.test.js]: empty line

**Comment by gthecht** [comment]
Where is your:
- .gitignore
- README
- package-lock

Add the gitignore and README in main, and the package lock where you have added your dependencies.

In addition, remove the TEMP branch from GitHub, use it only on your personal PC

File: package.json [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+  "description": "",
+  "main": "dist/index.js",
+  "scripts": {
+    "test": "jest --collectCoverage",
+    "start": "ts-node server.js"   
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "body-parser": "^1.20.1",
```
Comment (gthecht) [inline package.json:14]: unused

File: package.json [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+    "cors": "^2.8.5",
```
Comment (gthecht) [inline package.json]: unused

File: package.json [hunk 28]
```diff
@@ -0,0 +1,38 @@
...
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "helmet": "^6.0.1",
+    "mongo": "^0.1.0",
+    "mongoose": "^6.9.1",
```
Comment (gthecht) [inline package.json:19]: unused

File: package.json [hunk 29]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "body-parser": "^1.20.1",
+    "cors": "^2.8.5",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "helmet": "^6.0.1",
+    "mongo": "^0.1.0",
```
Comment (gthecht) [inline package.json:18]: unused

File: package.json [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+  "scripts": {
+    "test": "jest --collectCoverage",
+    "start": "ts-node server.js"   
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "body-parser": "^1.20.1",
+    "cors": "^2.8.5",
+    "dotenv": "^16.0.3",
```
Comment (gthecht) [inline package.json:15]: unused

**Comment by danielbilikId** [comment]
Updated & final version of Task_1

File: package.json (hunk 17 above)
Comment (gthecht) [reply package.json]: There are better ways to test jest...

Try [ts-jest](https://kulshekhar.github.io/ts-jest/docs/)

[here's why](https://kulshekhar.github.io/ts-jest/docs/babel7-or-ts)

**Review by gthecht — CHANGES_REQUESTED** [review]

File: package.json (hunk 17 above)
Comment (danielbilikId) [reply package.json]: Ok il add it in the next task

File: README.md (hunk 20 above)
Comment (danielbilikId) [reply README.md:1]: Il add it .

File: package.json (hunk 17 above)
Comment (gthecht) [reply package.json]: No add it here
